File: exCeption/exCeption/src/main/java/tn/capgemini/stackquestion/dto/KnowledgeDTO.java
package tn.capgemini.stackquestion.dto;

import lombok.Data;

import java.util.Date;

@Data
public class KnowledgeDTO {
    private Integer id;
    private String title;
    private String description;
    private String content;
    private String imageBase64;
    private String imageType;
    private String fileBase64;
    private String fileType;
    private String departement;
    private String projet;
    private Integer userId;
    private String userName;
    private Date createdDate;
    private int voteCount;

}

File: frontpfe/src/app/components/profile/Knowledge.dto.ts
export interface KnowledgeDTO {
    id: number;
    title: string;
    description: string;
    content: string;
    departement: string;
    projet: string;
    userId: number;
    userName?: string;
    imageBase64?: string;
    imageType?: string;
    fileBase64?: string;
    fileType?: string;
    createdDate?: string;
    voteCount?: number;
  }
  